### ISSUE TITLE
memcpy in SetFromKey might overflow the underling buffer

### ISSUE THREAD

**Issue by Novemser** [cmu-db/bustub#209]
`SetFromKey` in `generic_key.h`  performs memcpy to the underlying `data_` array. However, it seems to me that if `tuple.GetLength() > KeySize`, the memory could be corruptted.
```
inline void SetFromKey(const Tuple &tuple) {
  // intialize to 0
  memset(data_, 0, KeySize);
  memcpy(data_, tuple.GetData(), tuple.GetLength()); // if tuple.GetLength() > KeySize, the memory could be corruptted
}
...
// actual location of data, extends past the end.
char data_[KeySize];
```
I encountered a `*** stack smashing detected ***` error and found it might stem from the above issue.

If I understand correctly, the memcpy should be something like:
```
memcpy(data_, tuple.GetData(), KeySize);
```

Please correct me if I'm somehow wrong. Thanks : )

**Comment by Drinkwater-alt** [comment]
in the case tuple.GetLength() greater than KeySize will occur the error. I met the same error when doing project#3, and I found that I didn't convert the schema of the original tuple to output_schema in seq_scan_executor. I fixed that and the error disappears.

assume there are schema1 (col_A, col_B) and schema2(col_A). if a update_executor takes tuples from a child_seq_scan_executor, update_executor expects for a tuple with schema2 but the child_executor returns a tuple with schema1, the error would encounter when update_executor call SetFromKey for the latter has a larger size than the previous one.
I think it's a potential rule that we should make sure KeySize always equals to GetLength.
